src/mean.jl

```diff
@@ -29,99 +9,139 @@ julia> srs = SimpleRandomSample(apisrs;popsize=:fpc);
-    nsdh = []
-    substrata_domain_totals = []
-    Nh = []
-    nh = []
-    fh = []
-    ȳsdh = []
-    sigma_ȳsh_squares = []
-    grouped_frame = groupby(df, :strata)
-    for each_strata in keys(grouped_frame)
-        nsh = nrow(grouped_frame[each_strata])#, nrow=>:nsdh).nsdh
-        push!(nsdh, nsh)
-        substrata_domain_total = sum(grouped_frame[each_strata].x)
-        ȳdh = Statistics.mean(grouped_frame[each_strata].x)
-        push!(ȳsdh, ȳdh)
-        push!(substrata_domain_totals, substrata_domain_total)
-        popsizes = first(grouped_frame[each_strata].popsize)
-        push!(Nh, popsizes)
-        sampsizes = first(grouped_frame[each_strata].sampsize)
-        push!(nh, sampsizes)
-        sampfrac = first(grouped_frame[each_strata].sampfraction)
-        push!(fh, sampfrac)
-        push!(sigma_ȳsh_squares, sum((grouped_frame[each_strata].x .- ȳdh) .^ 2))
+function mean(x::Symbol, by::Symbol, design::StratifiedSample)
+    function domain_mean(x::AbstractVector, popsize::AbstractVector, sampsize::AbstractVector, sampfraction::AbstractVector, strata::AbstractVector)
+        df = DataFrame(x=x, popsize=popsize, sampsize=sampsize, sampfraction=sampfraction, strata=strata)
+        function calculate_components(x, popsize, sampsize, sampfraction)
+            return DataFrame(nsdh = length(x), nsh = length(x), substrata_domain_totals = sum(x), ȳsdh = mean(x), Nh = first(popsize), nh = first(sampsize),fh = first(sampfraction), sigma_ȳsh_squares = sum((x .- mean(x)).^2))
+        end
+        components = combine(groupby(df, :strata), [:x, :popsize, :sampsize, :sampfraction] => calculate_components => AsTable)
+        domain_mean = sum(components.Nh .* components.substrata_domain_totals ./ components.nh) / sum(components.Nh .* components.nsdh ./ components.nh)
+        pdh = components.nsdh ./ components.nh
+        N̂d = sum(components.Nh .* pdh)
+        domain_var = sum(components.Nh .^ 2 .* (1 .- components.fh) .* (components.sigma_ȳsh_squares .+ (components.nsdh .* (1 .- pdh) .* (components.ȳsdh .- domain_mean) .^ 2)) ./ (components.nh .* (components.nh .- 1))) ./ N̂d .^ 2
+        domain_mean_se = sqrt(domain_var)
+        return DataFrame(mean=domain_mean, SE=domain_mean_se)
     end
-    domain_mean = sum(Nh .* substrata_domain_totals ./ nh) / sum(Nh .* nsdh ./ nh)
-    pdh = nsdh ./ nh
-    N̂d = sum(Nh .* pdh)
-    domain_var = sum(Nh .^ 2 .* (1 .- fh) .* (sigma_ȳsh_squares .+ (nsdh .* (1 .- pdh) .* (ȳsdh .- domain_mean) .^ 2)) ./ (nh .* (nh .- 1))) ./ N̂d .^ 2
-    domain_mean_se = sqrt(domain_var)
-    return DataFrame(domain_mean=domain_mean, domain_mean_se=domain_mean_se)
+    gdf_domain = groupby(design.data, by)
+    combine(gdf_domain, [x, :popsize,:sampsize,:sampfraction, design.strata] => domain_mean => AsTable)
 end
-
 """
 Estimate the population mean of a variable of a stratified sample, and the corresponding standard error.
```

Comment:
Again this mean should not be documented, since it's the same functionality as above (starting with line 1). And I think this should be grouped with the first two methods.